src/main/java/com/minecolonies/structures/client/TemplateTessellator.java

```diff
@@ -0,0 +1,147 @@
+package com.minecolonies.structures.client;
+
+import com.minecolonies.structures.lib.RenderUtil;
+import net.minecraft.client.Minecraft;
+import net.minecraft.client.renderer.*;
+import net.minecraft.client.renderer.texture.ITextureObject;
+import net.minecraft.client.renderer.vertex.DefaultVertexFormats;
+import net.minecraft.client.renderer.vertex.VertexBuffer;
+import net.minecraft.client.renderer.vertex.VertexFormatElement;
+import net.minecraft.util.Mirror;
+import net.minecraft.util.Rotation;
+import net.minecraft.util.math.BlockPos;
+
+import static org.lwjgl.opengl.GL11.GL_ONE_MINUS_SRC_ALPHA;
+import static org.lwjgl.opengl.GL11.GL_SRC_ALPHA;
+
+public class TemplateTessellator
+{
+
+    private final BufferBuilder builder;
+    private final VertexBuffer         buffer      = new VertexBuffer(DefaultVertexFormats.BLOCK);
+    private final VertexBufferUploader vboUploader = new VertexBufferUploader();
+    private       boolean              isReadOnly  = false;
+
+    public TemplateTessellator()
+    {
+        this.builder = new BufferBuilder(2097152);
+        this.vboUploader.setVertexBuffer(buffer);
+    }
+
+    /**
+     * Draws the data set up in this tessellator and resets the state to prepare for new drawing.
+     */
+    public void draw(final Rotation rotation, final Mirror mirror, final Vector3d drawingOffset, final BlockPos inTemplateOffset)
+    {
+        if (!isReadOnly)
+        {
+            this.builder.finishDrawing();
+            this.vboUploader.draw(this.builder);
+            this.isReadOnly = true;
+        }
+
+        preTemplateBufferBinding(rotation, mirror, drawingOffset, inTemplateOffset);
+
+        this.buffer.bindBuffer();
+
+        preTemplateDraw();
+
+        this.buffer.drawArrays(7);
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Assign this magic number 7 to a well-named constant, and use the constant instead. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)